python/lsst/ap/association/dia_object.py

```diff
@@ -132,29 +141,51 @@ def update(self):
 
         self._updated = False
 
-        # compute all summary statistics on the catalog of DIASources.
+        # To quickly compute the summary statistics we check if the catalog
+        # is currently contious and if not we make a deep copy.
+        if not self._dia_source_catalog.isContiguous():
+            tmp_dia_source_catalog = self._dia_source_catalog.copy(deep=True)
+            del self._dia_source_catalog
+            self._dia_source_catalog = tmp_dia_source_catalog
 
-        self.compute_summary_statistics()
+        self._compute_summary_statistics()
 
         self._updated = True
 
         return None
 
-    @property
-    def is_updated(self):
-        """ Return the current state of this DIAObject.
-
-        If True is returned this DIAObject has computed all of its summary
-        statistics for the current collection of DIASources that make it up.
+    def _compute_summary_statistics(self):
+        """ Retrive properties from DIASourceCatalog attribute and update the
+        summary statistics that represent this DIAObject
 
-        Return
-        ------
-        bool
+        Returns
+        -------
+        None
         """
 
-        return self._updated
+        # Loop through DIASources, compute summary statistics (TBD) and store
+        # them in dia_object_record attribute.
+
+        for name in self.schema.getNames():
```

Comment:
Removed most of the functionality here and into it's own function _compute_mean_coordinate. Computation of errors has been moved to another ticket.